### PR TITLE
Add analysis about number of addons per user

### PR DESCRIPTION
**Pull request by florian**
I added a new section to the notebook that displays some information about the number of addons per user.

### CODE REVIEW

File: evaluation/evaluation.md [hunk 1]
```diff
@@ -1764,3 +1768,328 @@ display_merged_filtered_results(counts, titles, total_results, combinations, lab
...
+def plot_addon_distribution(addon_counts):
+    num_addons, num_addons_count = addon_counts_to_proportions(addon_counts)
+    
+    plt.bar(num_addons, num_addons_count, width=1.)
+    plt.title("Number of addons per user")
+    plt.xlabel("Number of addons")
+    plt.ylabel("Proportion of users")
+    plt.show()
+    
+    plt.plot(np.cumsum(num_addons_count))
```
Comment (mlopatka) [inline evaluation/evaluation.md:1826]: Lets truncate the x axis at 20-30 ish and get a better resolution on the x-axis ticks (with a grid) to make this more clear.

File: evaluation/evaluation.md [hunk 2]
```diff
@@ -1764,3 +1768,328 @@ display_merged_filtered_results(counts, titles, total_results, combinations, lab
 </table>
 </div>
 
+
+## Addon counts
+
+We want to train an ensemble model using the individual recommenders that we already have. To optimize this ensemble model, we need some training data. We have information about what addons different users have installed, so it would make sense to check if our ensemble model would also recommend these addons to the respective users.
+
+However, there is one fundamental conflict here: To be able to make recommendations, the collaborative recommender already needs some information about which addons a user has installed. Thus, we can only use a subset of a user's installed addons for evaluation. These addons are masked and the collaborative recommender then only uses the unmasked addons. The open question is how large this subset of masked addons should be.
```
Comment (mlopatka) [inline evaluation/evaluation.md:1776]: I think the question is more broad than just the "size" of the subset. More generally, we need to make an informed decision on the inclusion/selection criteria of the validation set given the optimization paradigm we have designed.

**Review by mlopatka — CHANGES_REQUESTED** [review]
I think we need to refactor this repo to clean it up. the current notebooks are too large and contain too many things and should be split into smaller notebooks that each accomplish a specific sub-task.

File: evaluation/evaluation.md (hunk 1 above)
Comment (florian) [reply evaluation/evaluation.md:1826]: The last plot in the notebook has the x-axis truncated at 15 or 20. Is this already what you're looking for?

**Review by mlopatka — APPROVED** [review]

**Pull request merged**